zend_abstract_interface/env/env.c

```diff
@@ -0,0 +1,56 @@
+#include "env.h"
+
+#include <main/SAPI.h>
+#include <main/php.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if PHP_VERSION_ID >= 80000
+#define sapi_getenv_compat(name, name_len) sapi_getenv(name, name_len)
+#elif PHP_VERSION_ID >= 70000
+#define sapi_getenv_compat(name, name_len) sapi_getenv((char *)name, name_len)
+#else
+#define sapi_getenv_compat(name, name_len) sapi_getenv((char *)name, name_len TSRMLS_CC)
+#endif
+
+zai_env_result zai_getenv(const char *name, size_t name_len, char *buf, size_t buf_size) {
+#if PHP_VERSION_ID < 70000
+    TSRMLS_FETCH();
+#endif
+    if (!buf || !buf_size) return ZAI_ENV_ERROR;
+
+    buf[0] = '\0';
+
+    if (!name || !name_len) return ZAI_ENV_ERROR;
+
+    /* Some SAPIs do not initialize the SAPI-controlled environment variables
+     * until the request has started. It is for this reason we cannot reliably
+     * access environment variables outside of a request context.
+     */
+    if (!PG(modules_activated)) return ZAI_ENV_NOT_READY;
```

Comment:
Wow! Very nice catch. The very use-case for this API (first-time RINIT) won't even work as it is designed. 🤦 I'll make sure to change this and add a test that executes during RINIT.

Reply:
I've applied all of the suggestions except this one since it requires an update to the ZAI SAPI API that will allow us to add an "inline module" to test that we can access the environment variables in module RINIT. I started on the API, but I'm going to wrap it up tomorrow morning.

Reply:
This has been addressed in 20b1eb3. Thanks again for catching this one!